pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -80,6 +80,11 @@ const (
 	gceComputeAPIEndpoint = "https://www.googleapis.com/compute/v1/"
 )
 
+// APIObject is an abstraction of all GCE API object in go client
+type APIObject interface {
```

Comment:
Yes. It has to be exported. Ingress has a bunch of interface for GCE cloud provider.